Follow up to #9581

bangerth opened the pull request:
This is the promised changelog entry for #9581. I didn't think anything really needed to be updated in the readme file (nothing actively seemed wrong with the old text), but I always find something anyway.

kronbichler commented on doc/news/changes/minor/20200301Bangerth:
```diff
@@ -0,0 +1,8 @@
+New: The underlying type for types::global_dof_index used to be
+`unsigned long long int`. On all reasonable systems, this is actually
+a 64-bit unsigned integer type, but the name wouldn't tell you that
+unless you already knew. This is now fixed: Instead of the mouthfull
```
```suggestion
unless you already knew. This is now fixed: Instead of the mouthful
```

kronbichler commented on doc/readme.html:
```diff
@@ -336,10 +336,19 @@ <h4>Selecting optional library features</h4>
                 <code>-DDEAL_II_WITH_64BIT_INDICES=ON</code> argument to
-                <code>cmake</code>. You will not be able to solve problems of this size on a single machine, but only when using clusters of computers and linear algebra packages PETSc or Trilinos. To use this option with PETSc, PETSc must be compiled
+                <code>cmake</code>. You will not be able to solve
+                problems of this size on a single machine, but only
```
I don't think that is true any more. If you have a system with 512 GB of RAM, which is a bit more than the regular workstation/cluster but not that unusual either, you can solve systems of 6-8 billion dofs with explicit time integrators like the one from step-67 (#8319). I suggest to insert `typically`.

bangerth replied:
Good point. I was going to add "unless you have exceptional amounts of memory and patience", but I'm sure that even that will become moot over time :-)

kronbichler commented on doc/readme.html:
```diff
@@ -336,10 +336,19 @@ <h4>Selecting optional library features</h4>
+                <code>cmake</code>. You will not be able to solve
+                problems of this size on a single machine, but only
+                when using clusters of computers and linear algebra
+                packages PETSc or Trilinos. To use this option with
```
It also works with deal.II's own parallel vectors.

bangerth replied:
I just removed that whole reference to PETSc and Trilinos.

bangerth commented:
@kronbichler -- I've made the changes, and also:
* Moved the changelog entry to `incompatibilities`
* Added a paragraph that explained that strictly speaking, things are incompatible but that few places will notice (I sure hope that the latter is true).
